internal/services/containerapps/container_app_environment_managed_certificate_resource.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package containerapps
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerapps/2024-03-01/managedenvironments"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ContainerAppEnvironmentManagedCertificateResource struct{}
+
+var _ sdk.Resource = ContainerAppEnvironmentManagedCertificateResource{}
+
+type ContainerAppEnvironmentManagedCertificateModel struct {
+	Name                    string                 `tfschema:"name"`
+	ManagedEnvironmentId    string                 `tfschema:"container_app_environment_id"`
```

Comment:
Could we rename this to
```suggestion
	ContainerAppEnvironmentId    string                 `tfschema:"container_app_environment_id"`
```